Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1552,7 +1553,19 @@ void FrameLoader::loadURL(FrameLoadRequest&& frameLoadRequest, const String& ref
     action.setShouldReplaceDocumentIfJavaScriptURL(frameLoadRequest.shouldReplaceDocumentIfJavaScriptURL());
     action.setIsInitialFrameSrcLoad(frameLoadRequest.isInitialFrameSrcLoad());
     action.setNewFrameOpenerPolicy(frameLoadRequest.newFrameOpenerPolicy());
-    action.setNavigationAPIType(determineNavigationType(newLoadType, frameLoadRequest.navigationHistoryBehavior()));
+    auto historyHandling = frameLoadRequest.navigationHistoryBehavior();
+    bool isSameOrigin = frameLoadRequest.requesterSecurityOrigin().isSameOriginDomain(frame->document()->securityOrigin());
+    if (historyHandling == NavigationHistoryBehavior::Auto && !isReload(newLoadType)) {
+        RefPtr document = m_frame->document();
+        URL oldURL = document->url();
```

Comment:
Probably would read better without this local.

Reply:
Sure, fixed.